editor/src/components/canvas/canvas-strategies/strategies/absolute-duplicate-strategy.tsx

```diff
@@ -73,43 +77,46 @@ export function absoluteDuplicateStrategy(
         let duplicatedElementNewUids = {
           ...customStrategyState.duplicatedElementNewUids,
         }
-        let withDuplicatedMetadata: ElementInstanceMetadataMap = {
-          ...canvasState.startingMetadata,
-        }
-        let duplicateCommands: Array<DuplicateElement> = []
+        let duplicateCommands: Array<CanvasCommand> = []
         let newPaths: Array<ElementPath> = []
 
         filteredSelectedElements.forEach((selectedElement) => {
           const selectedElementString = EP.toString(selectedElement)
+          const oldUid = EP.toUid(selectedElement)
           const newUid =
             duplicatedElementNewUids[selectedElementString] ??
             generateUidWithExistingComponents(canvasState.projectContents)
-          const newPath = EP.appendToPath(EP.parentPath(selectedElement), newUid)
-          const newPathString = EP.toString(newPath)
-
           duplicatedElementNewUids[selectedElementString] = newUid
-          withDuplicatedMetadata[newPathString] = {
-            ...withDuplicatedMetadata[EP.toString(selectedElement)],
-            elementPath: newPath,
-          }
 
-          duplicateCommands.push(duplicateElement('always', selectedElement, newUid))
+          const newPath = EP.appendToPath(EP.parentPath(selectedElement), newUid)
+
           newPaths.push(newPath)
+          duplicateCommands.push(
+            duplicateElement('always', selectedElement, newUid, 'before'),
+            /**
+             * TODO: the UIDs are swapped to make it seem like that the element being dragged
+             * (the original element) is the new, duplicated element. While this works on the canvas,
+             * in the code editor, the wrong code snippet is highlighted until the affected files are saved.
+             *
+             * I added this as a fallback solution, definitely needs a better approach
+             */
+            updatePropIfExists('on-complete', selectedElement, create('data-uid'), newUid),
+            updatePropIfExists('on-complete', newPath, create('data-uid'), oldUid),
```

Comment:
This issue still appears to be present (I was just able to reproduce it on the current version of this branch). I don't think we need to (or even should) swap the UIDs here. As I mentioned in another comment, the UIDs are internal only anyway, so as long as the moved element is the one that appears second in the Navigator / in code then I think we're good tbh

Reply:
Removed the uid switching in https://github.com/concrete-utopia/utopia/pull/3359/commits/c7b7dd1730fe94b548303f68d595f0747b989934